### ISSUE TITLE
Running FastAPI within Django

### ISSUE THREAD

**Issue by jrharmon** [tiangolo/fastapi#2897]
Is there any official support for using FastAPI inside of a Django project?  Looking online, I see a few blog posts about various ways people got it to work, but none of them feel like something that is vetted, and would be production ready.

We use Django at my company for out services, and want to keep using it for its ORM, management commands, admin UI, etc.  Being able to replace DRF (which is sync-only) with FastAPI would be great for the API-layer.

If this is a common enough use-case, even a simple guide in the docs pointing out what is required, and any things to watch out for, would be great.

**Comment by Kludex** [comment]
WSGI can be considered as a subset of ASGI. 

This is the way to go: https://fastapi.tiangolo.com/advanced/wsgi/?h=wsgi#using-wsgimiddleware
In other words: Django inside FastAPI.

**Comment by Maxpowerfreak** [comment]
If you need another example, here's a basic Django/FastAPI app someone built using the concept from the documentation @Kludex linked above: https://github.com/jordaneremieff/django-fastapi-example

Personally, I would recommend having both systems separate. This would allow you to scale them independently and make it easier to develop/maintain as you would not have to deploy the full app every time you update one or the other.

**Comment by Kludex** [comment]
The way it's recommended: if you have a large WSGI project, then you can migrate to FastAPI step by step by mounting one application into the other.

There's nothing wrong with that. 😗

**Comment by jrharmon** [comment]
Thanks for the links.  Mounting FastAPI inside of Django would probably be an easier sell than FastAPI being just for some routes inside of a normal Django app.  I actually found the [Django-Ninja API](https://django-ninja.rest-framework.com/), which is heavily inspired by FastAPI, but is built to integrate seamlessly with Django.  It works great so far.

**Comment by Pei2tech** [comment]
We are also looking for a solution for our Django project. If FastAPI can support Django would be a great idea. However, I will test the Django-Ninja API. Thank @jrharmon 's link.

**Comment by tiangolo** [comment]
Yep, you could mount a Django app inside of a FastAPI app, but that would probably be more difficult to debug. I would recommend that you deploy both the Django and FastAPI apps independently and route to each one based on a path prefix or something similar. E.g. `/api/v1` goes to Django and `/api/v2` goes to FastAPI.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
